src/Azure.Deployments.Extensibility.Core/V2/IExtensibilityProvider.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Deployments.Extensibility.Core.V2.Models;
+using Microsoft.AspNetCore.Http;
+using System.Text.Json.Nodes;
+
+namespace Azure.Deployments.Extensibility.Core.V2
+{
+    public interface IExtensibilityProvider
```

Comment:
For our current hosting model, all extensibility providers run in the context of the extensibility host app. The extensibility host app handles authentication and authorization for all extensible resource requests from ARM. It then calls extensibility providers to process the requests, which may require another layer of authentication and authorization. However, depending on the underlying API implementation, providers may use different authentication and authorization mechanisms. For the k8s provider, kubeconfig is used, which is part of the request body. For the Graph provider, the Authorization header is used, which is accessible via `HttpContext` (the first parameter).

Reply:
The standard way is to use OAuth OBO, and I believe the `Authorization` header is the credentials parameter. The k8s provider is just an outlier here, since it cannot leverage OAuth OBO. I'm not sure about the security risk part since the payload won't be logged anywhere and it's short lived.